### ISSUE TITLE
Repeatedly hitting 'next' or 'back' in a playlist causes UI to freeze and possibly crash DeaDBeeF

### ISSUE THREAD

**Issue by formirthonly** [cboxdoerfer/ddb_waveform_seekbar#14]
Thank you for this wonderful plugin - without it I might never have used DeaDBeeF! I'm currently using version 0.7.2 of DeaDBeeF built from source, and 0.5 of ddb_waveform_seekbar also built from source.

So the problem I'm having is that DeaDBeeF crashes pretty quickly if you hold down the hotkey for 'next' or 'back'. If you tap the keys in quick succession, it will freeze up the UI while playing one of the tracks that was next in the queue. When I remove the plugin, the problem goes away and DeaDBeeF will handle holding loading tracks perfectly.

**Comment by cboxdoerfer** [comment]
Good catch, thx for reporting. There are a couple of issues here which need fixing. Most of them can probably be fixed with a more intelligent queue management for decoding. I'll have a look at that right away.
